cmd/ctr/commands/containers/create.go

```diff
@@ -0,0 +1,142 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package containers
+
+import (
+	gocontext "context"
+	"encoding/csv"
+	"fmt"
+	"strings"
+
+	"github.com/containerd/containerd/cmd/ctr/commands"
+	"github.com/containerd/containerd/containers"
+	"github.com/containerd/containerd/errdefs"
+	clabels "github.com/containerd/containerd/labels"
+	"github.com/containerd/containerd/oci"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+)
+
+var createCommand = cli.Command{
+	Name:           "create",
+	Usage:          "create container",
+	ArgsUsage:      "[flags] Image|RootFS CONTAINER [COMMAND] [ARG...]",
+	SkipArgReorder: true,
+	Flags:          append(append(commands.SnapshotterFlags, []cli.Flag{commands.SnapshotterLabels}...), commands.ContainerFlags...),
+	Action: func(context *cli.Context) error {
+		var (
+			id     string
+			ref    string
+			config = context.IsSet("config")
+		)
+
+		if config {
+			id = context.Args().First()
+			if context.NArg() > 1 {
+				return fmt.Errorf("with spec config file, only container id should be provided: %w", errdefs.ErrInvalidArgument)
+			}
+		} else {
+			id = context.Args().Get(1)
+			ref = context.Args().First()
+			if ref == "" {
+				return fmt.Errorf("image ref must be provided: %w", errdefs.ErrInvalidArgument)
+			}
+		}
+		if id == "" {
+			return fmt.Errorf("container id must be provided: %w", errdefs.ErrInvalidArgument)
+		}
+		client, ctx, cancel, err := commands.NewClient(context)
+		if err != nil {
+			return err
+		}
+		defer cancel()
+		_, err = NewContainer(ctx, client, context)
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+}
+
+func withMounts(context *cli.Context) oci.SpecOpts {
```

Comment:
Is this more suited in `ctr/commands/containers/containers.go`. I assume its not directly used by any function that is specific to creating a container.

Reply:
You are right, it should be in container.go, which is used by NewContainer in container_*.go

Reply:
@akhilerm updated, PTAL